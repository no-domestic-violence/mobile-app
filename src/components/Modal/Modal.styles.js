import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  textSuccess: {
    color: 'darkgreen',
  },
  modalContainer: {
    backgroundColor: '#f9fafb',
    width: '80%',
    borderRadius: 5,
    alignContent: 'center',
    justifyContent: 'center',
    padding: 20,
  },
});
