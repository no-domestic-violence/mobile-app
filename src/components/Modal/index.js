import React from 'react';
import Modal from 'react-native-modal';
import { View, Text, Button } from 'react-native';
import { styles } from './Modal.styles';

const ModalComponent = ({ isVisible, setModalVisible, navigation }) => {
  return (
    <Modal isVisible={isVisible}>
      <View
        style={{
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <View style={styles.modalContainer}>
          <Text style={styles.textSuccess}>
            Your password was successfully changed!
          </Text>
          <Button
            title="Ok"
            onPress={() => {
              setModalVisible(false);
              navigation.navigate('User');
            }}
          />
        </View>
      </View>
    </Modal>
  );
};

export default ModalComponent;
