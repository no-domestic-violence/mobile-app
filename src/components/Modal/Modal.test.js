import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Modal from './index';


describe('Modal component', () => {
  const setModalVisible = jest.fn();
  const mockNavigation = {
    navigate: jest.fn(),
  };

  const renderComponent = () => {
    return (
      <Modal
        isVisible={true}
        setModalVisible={setModalVisible}
        navigation={mockNavigation}
      />
    );
  };

  it('should render modal when it is visible', () => {
    const { getByText } = render(renderComponent());
    expect(getByText(/password was successfully changed/i)).not.toBeNull();
  });
  it('should set visibility of modal to false on press and navigate to User', () => {
    const { getByText } = render(renderComponent());
    fireEvent.press(getByText(/Ok/i));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('User');
    expect(setModalVisible).toHaveBeenCalledTimes(1);
    expect(setModalVisible).toHaveBeenCalledWith(false);
  });
  it('should match snapshot', () => {
    const result = render(renderComponent()).toJSON();
    expect(result).toMatchSnapshot();
  });
});
