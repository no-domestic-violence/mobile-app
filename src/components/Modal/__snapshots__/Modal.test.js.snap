// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should match snapshot 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#f9fafb",
            "borderRadius": 5,
            "justifyContent": "center",
            "padding": 20,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "darkgreen",
            }
          }
        >
          Your password was successfully changed!
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Ok
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
